conformance/lang/values-types-and-variables/array.balt

```diff
@@ -0,0 +1,99 @@
+Test-case: output
+Description: Test valid array types and values
+Label: array
+
+const LENGTH = 2;
+
+public function main() returns () {
```

Comment:
This test is mixing a lot of different things each of which require different features. I think tests should be more fine-grained.

Reply:
Specifically, separate tests for

- array of int (just plain int[])
    - should test mutation of the array (including just past the length and with gaps)
    - should test the length
- array of boolean
- array of string
- array of float
- array of any
- array of byte
- array of xml
- fixed length arrays of any of the above (the above should be variable length)
- using `*` as the length of a fixed length array
- tests for subtype relationships between arrays
- tests for inherent type violations

Reply:
+1. This covers lot of cases. Should be divided into multiple test-cases.

Reply:
Will divide